coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,680 @@
+                    - end_line - The line where the issue ends.
+                    - end_column - The column where the issue ends.
+                    - severity - The severity of the issue.
+                    - message - The message of the result.
+                    - origin - The origin of the issue.
+
+                    The groups ``line``, ``column``, ``end_line`` and
+                    ``end_column`` don't have to match numbers only, they can
+                    also match nothing, the generated ``Result`` is filled
+                    automatically with ``None`` then for the appropriate
+                    properties.
+                :param severity_map:
+                    A dict used to map a severity string (captured from the
+                    ``output_regex`` with the named group ``severity``) to an
+                    actual ``coalib.results.RESULT_SEVERITY`` for a result.
+                :return:
+                    An iterator returning results.
+                """
+                if isinstance(output, str):
+                    output = (output,)
+
+                for string in output:
+                    for match in re.finditer(output_regex, string):
+                        yield self._convert_output_regex_match_to_result(
+                            match, filename, severity_map=severity_map)
+
+            if options["output_format"] is None:
+                # Check if user supplied a `process_output` override.
+                if not (hasattr(klass, "process_output") and
+                        callable(klass.process_output)):
+                    raise ValueError("`process_output` not provided by given "
+                                     "class.")
+                # No need to assign to `process_output` here, the class mixing
+                # below automatically does that.
+            else:
+                # Prevent people from accidentally defining `process_output`
+                # manually, as this would implicitly override the internally
+                # set-up `process_output`.
+                if hasattr(klass, "process_output"):
+                    raise ValueError("`process_output` is used by given class,"
+                                     " but " + repr(options["output_format"]) +
+                                     " output format was specified.")
+
+                if options["output_format"] == "corrected":
+                    process_output_args = {}
+                    if "diff_severity" in options:
+                        process_output_args["diff_severity"] = (
+                            options["diff_severity"])
+                    if "diff_message" in options:
+                        process_output_args["diff_message"] = (
+                            options["diff_message"])
+
+                    process_output = partialmethod(
+                        process_output_corrected, **process_output_args)
```

Comment:
Easier to read:

```
process_output_args = {
    k:v for k, v in options.items()
        if k in ("diff_message", "diff_severity")}
```
 - Removes writing "diff_message" 3 times
 - Removed multiple if conditions

BTW: You seem to be doing similar things freqquently (Like L285). Consider moving this into a function like `splice_dict(dictionary, keys)` or so ?